feat: Updated adxl345 sensor driver to configuration via devicetree file.

whati001 opened the pull request:
Followup PR of: #66046 

Updated/rewritten adxl345 sensor code to support devicetree parameters. I have tested the code with an nrf52840-dk and a adxl345 connected via I2C.

yperess commented on drivers/sensor/adxl345/adxl345.c:
```diff
@@ -63,50 +63,59 @@ static int adxl345_reg_access_spi(const struct device *dev, uint8_t cmd, uint8_t
 }
 
-static inline int adxl345_reg_write_byte(const struct device *dev, uint8_t addr, uint8_t val)
+int adxl345_reg_write_byte(const struct device *dev, uint8_t addr, uint8_t val)
```
nit: leave `static`

whati001 replied:
added `static`

yperess commented on drivers/sensor/adxl345/adxl345.c:
```diff
@@ -63,50 +63,59 @@ static int adxl345_reg_access_spi(const struct device *dev, uint8_t cmd, uint8_t
 
-static inline int adxl345_reg_read_byte(const struct device *dev, uint8_t addr, uint8_t *buf)
-
+int adxl345_reg_read_byte(const struct device *dev, uint8_t addr, uint8_t *buf)
```
nit: leave `static`

yperess commented on drivers/sensor/adxl345/adxl345.c:
```diff
@@ -116,40 +125,36 @@ static int adxl345_read_sample(const struct device *dev,
+		}
 	}
 
 	__ASSERT_NO_MSG(samples_count <= ARRAY_SIZE(data->bufx));
```
I noticed the following in the datasheet:

> there must be at least 5 µs between the end of reading
> the data registers and the start of a new read of the FIFO or a read
> of the FIFO_STATUS register (Address 0x39). The end of reading
> a data register is signified by the transition from Register 0x37 to
> Register 0x38 or by the CS pin going high.

The loop below might need to be tweaked to verify this delay.

whati001 replied:
The loop will be removed via #67499 . Added temporarily a sleep for 5 us.

MaureenHelm commented on drivers/sensor/adxl345/adxl345.c:
```diff
@@ -255,34 +353,41 @@ static int adxl345_init(const struct device *dev)
+				     &adxl345_config_##inst, POST_KERNEL,                          \
+				     CONFIG_SENSOR_INIT_PRIORITY, &adxl345_api_funcs);
+
+DT_INST_FOREACH_STATUS_OKAY(ADXL345_DEFINE)
```
This looks like a lot of whitespace noise. Please keep major whitespace reformatting changes in separate commits from functional changes

whati001 replied:
Refactored code to reduce whitespace noise.

MaureenHelm reviewed (changes requested):
Thanks for your contribution!

Please address compliance failures and review feedback by amending commits instead of adding fixup commits.

whati001 commented:
Updated code and normally fixed compliance checks. Please apologize, but I am not sure why the last three checks are not running.

whati001 commented:
@MaureenHelm please provide an update

MaureenHelm commented on drivers/sensor/adxl345/adxl345.c:
```diff
@@ -63,50 +63,59 @@ static int adxl345_reg_access_spi(const struct device *dev, uint8_t cmd, uint8_t
 }
 
-static inline int adxl345_reg_write(const struct device *dev, uint8_t addr, uint8_t *data,
-				    uint8_t len)
+static int adxl345_reg_write(const struct device *dev, uint8_t addr, uint8_t *data, uint8_t len)
 {
-
 	return adxl345_reg_access(dev, ADXL345_WRITE_CMD, addr, data, len);
 }
 
-static inline int adxl345_reg_read(const struct device *dev, uint8_t addr, uint8_t *data,
-				   uint8_t len)
+static int adxl345_reg_read(const struct device *dev, uint8_t addr, uint8_t *data, uint8_t len)
 {
-
 	return adxl345_reg_access(dev, ADXL345_READ_CMD, addr, data, len);
 }
 
-static inline int adxl345_reg_write_byte(const struct device *dev, uint8_t addr, uint8_t val)
+static int adxl345_reg_write_byte(const struct device *dev, uint8_t addr, uint8_t val)
 {
 	return adxl345_reg_write(dev, addr, &val, 1);
 }
 
-static inline int adxl345_reg_read_byte(const struct device *dev, uint8_t addr, uint8_t *buf)
-
+static int adxl345_reg_read_byte(const struct device *dev, uint8_t addr, uint8_t *buf)
```
These changes aren't related to `feat: Add devicetree support for adxl345 sensor`. Please revert

whati001 replied:
reverted white space noise as good as possible.

MaureenHelm commented on drivers/sensor/adxl345/adxl345.c:
```diff
@@ -116,40 +125,37 @@ static int adxl345_read_sample(const struct device *dev,
-		LOG_ERR("Failed to read FIFO status rc = %d\n", rc);
-		return rc;
+
+	if (data->fifo_mode == ADXL345_FIFO_BYPASS) {
+		samples_count = 1;
+	} else {
+		rc = adxl345_reg_read_byte(dev, ADXL345_FIFO_STATUS_REG, &samples_count);
+		if (rc < 0) {
+			LOG_ERR("Failed to read FIFO status rc = %d\n", rc);
+			return rc;
+		}
```
Also not related to devicetree. Please revert

whati001 replied:
This change is needed to fetch samples properly if FIFO is disabled.

Because if FIFO is disabled, reading from ADXL345_FIFO_STATUS_REG returns 0.
If sample_count is zero, the driver will never fetch a sample within the loop body.
![image](https://github.com/zephyrproject-rtos/zephyr/assets/16085873/ba5fd589-3272-4542-938a-3ddb5b4a022e)

whati001 replied:
This behavior will be fixed with #67499

MaureenHelm commented on drivers/sensor/adxl345/adxl345.c:
```diff
@@ -163,13 +169,14 @@ static int adxl345_sample_fetch(const struct device *dev,
 		data->bufy[s] = sample.y;
 		data->bufz[s] = sample.z;
+
+		k_usleep(ADXL345_SAMPLE_SLEEP_TIME_US);
```
Same

MaureenHelm commented on dts/bindings/sensor/adi,adxl345-common.yaml:
```diff
@@ -0,0 +1,46 @@
+    type: int
+    default: 100
+    description: |
+          Accelerometer sampling frequency (ODR). Default is 100Hz.
```
Need to justify the default value in the description. Is it a register reset value?

whati001 replied:
Added justification.

MaureenHelm commented on dts/bindings/sensor/adi,adxl345-common.yaml:
```diff
@@ -0,0 +1,46 @@
+    type: int
+    default: 16
+    description: |
+      Accelerometer sampling range in G
```
Need to justify the default value in the description. Is it a register reset value?

whati001 replied:
Updated default value to reset value and updated description text.

whati001 commented:
@MaureenHelm please apologize to bother you again, but I have now reverted all whitespace issues as good as possible and one change is mandatory to ensure functionality if FIFO is disabled (possible now via DT setup).
Maybe you can catch some time to review one more time :)

whati001 commented:
Rebased on current main

MaureenHelm reviewed (changes requested):
> @MaureenHelm please apologize to bother you again, but I have now reverted all whitespace issues as good as possible and one change is mandatory to ensure functionality if FIFO is disabled (possible now via DT setup).
Maybe you can catch some time to review one more time :)

Thank you, but these changes need to be amended to the original commits. In other words, don't introduce all the whitespace changes in one commit and then revert them in another commit; avoid introducing the whitespace changes in the first place. Similarly, amend the commit that introduces the new devicetree properties to include their default justifications.

whati001 commented:
Hi @MaureenHelm 

thank you very much for your feedback and please apologize the circumstances.
As requested, I have squashed all the commits and force pushed to includes everything within a single commit without "reverting" commits.

whati001 commented:
Fixed compliance issues, please review again @MaureenHelm

whati001 committed:
feat: Extended adxl345 sensor devicetree file definition

Added sampling frequency, range, and fifo mode configuration
to adxl345 sensor devicetree file.

Signed-off-by: Andreas Karner <andreaskarner@outlook.com>

whati001 committed:
feat: Add devicetree support for adxl345 sensor

Rewritten adxl345 sensor driver to accept configuration
via devicetree system.

Signed-off-by: Andreas Karner <andreaskarner@outlook.com>
